tensorflow/compiler/xla/service/llvm_ir/ir_array.cc

```diff
@@ -198,12 +198,14 @@ IrArray::Index IrArray::Index::SourceIndexOfReshape(
     // 'source_multidim_index' which does not belong to 'deleted_dims_indices',
     // we retrieve the corresponding value from 'multidim_' (skipping any
     // indices that appear in 'inserted_dims_indices').
-    for (int64 i = 0, j = 0, k = 0, l = 0; i < source_multidim_index.size();
-         ++i) {
-      if (j == deleted_dims_indices.size() || deleted_dims_indices[j] > i) {
+    for (int64 i = 0, j = 0, k = 0, l = 0, 
+         end = source_multidim_index.size(); i < end; ++i) {
```

Comment:
Move the `end` initialization to the line above, to fit with the other initializations